cyral/internal/policyv2/resource.go

```diff
@@ -0,0 +1,152 @@
+			},
+			"enabled": {
+				Description: "Indicates if the policy is enabled.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"scope": {
+				Description: "Scope of the policy. If empty or omitted, all repositories are in scope.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"repo_ids": {
+							Description: "List of repository IDs that are in scope.",
+							Type:        schema.TypeList,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Optional:    true,
+						},
+					},
+				},
+			},
+			"tags": {
+				Description: "Tags associated with the policy to categorize it.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"valid_from": {
+				Description: "Time when the policy comes into effect. If omitted, the policy is in effect immediately.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"valid_until": {
+				Description: "Time after which the policy is no longer in effect. If omitted, the policy is in effect indefinitely.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"document": {
+				Description: "The actual policy document in JSON format. It must conform to the schema for the policy type.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"last_updated": {
+				Description: "Information about when and by whom the policy was last updated.",
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"created": {
+				Description: "Information about when and by whom the policy was created.",
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"enforced": {
+				Description: "Indicates if the policy is enforced. If not enforced, no action is taken based on the policy, but alerts are triggered for violations.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"type": {
+				Description: "Type of the policy, one of [`local`, `global`]",
+				Type:        schema.TypeString,
+				Required:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(string)
+					// Valid values for the type fields have aliases ie POLICY_TYPE_GLOBAL and global are the same.
+					// This is consistent with the API, which accepts both.
+					// However, the recommendation is to stick to [local, global, approval].
+					validTypes := []string{"POLICY_TYPE_GLOBAL", "global", "POLICY_TYPE_LOCAL", "local", "POLICY_TYPE_APPROVAL", "approval"}
+					for _, validType := range validTypes {
+						if v == validType {
+							return
+						}
+					}
+					errs = append(errs, fmt.Errorf("%q must be one of %v, got: %s", key, validTypes, v))
+					return
+				},
```

Comment:
For the `Description` and `ValidateFunc`, please see [this example](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/internal/repository/datasource.go#L87-L90). You can simplify the code by just providing a slice with the valid values in both assignments.